tachyon/crypto/commitments/fri/two_adic_fri_prover.h

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#ifndef TACHYON_CRYPTO_COMMITMENTS_FRI_TWO_ADIC_FRI_PROVER_H_
+#define TACHYON_CRYPTO_COMMITMENTS_FRI_TWO_ADIC_FRI_PROVER_H_
+
+#include <vector>
+
+#include "tachyon/base/bits.h"
+#include "tachyon/crypto/commitments/fri/two_adic_fri_config.h"
+#include "tachyon/crypto/commitments/fri/two_adic_fri_proof.h"
+#include "tachyon/math/finite_fields/extension_field_traits_forward.h"
+#include "tachyon/math/matrix/matrix_types.h"
+#include "tachyon/zk/air/plonky3/challenger/challenger.h"
+
+namespace tachyon::crypto {
+
+template <typename ExtField, typename ChallengeMMCS, typename Challenger>
+CommitPhaseResult<ExtField, ChallengeMMCS> CommitPhase(
+    TwoAdicFriConfig<ChallengeMMCS>& config,
+    const std::vector<std::vector<ExtField>>& inputs, Challenger& challenger) {
+  using Commitment = typename ChallengeMMCS::Commitment;
+  using ProverData = typename ChallengeMMCS::ProverData;
+
+  std::vector<ExtField> folded = inputs[0];
+  std::vector<Commitment> commits;
+  std::vector<ProverData> data;
+
+  while (folded.size() > config.Blowup()) {
+    math::RowMajorMatrix<ExtField> leaves((folded.size() + 1) / 2, 2);
+    OPENMP_PARALLEL_FOR(size_t i = 0; i < folded.size(); ++i) {
```

Comment:
I think it should be paralell depending on `folded.size()`

Reply:
What do you mean by this?